deploy/ansible/roles/process_bom/tasks/download_media.yml

```
@@ -0,0 +1,30 @@
+---
+
+- name: "Ensure media property is iterated"
+  vars:
+    archive_location: "{{ sapbits_location_base_path }}/{{ source_location }}"
+  block:
+    - name: Ensure target folders exist
+      file:
+        path: "{{ file_dest_target_location }}"
+        state: directory
+      vars:
+        file_dest_target_location: "{{ download_iterator.override_target_location is defined | ternary(download_iterator.override_target_location, default_location) | regex_replace('/$', '') }}"
+      loop: "{{ bom_media | flatten(levels=1) }}"
+      loop_control:
+        loop_var: download_iterator
```

I found a way of condensing the media list down to unique folders:

```suggestion
    - name: Ensure target folders exist
      file:
        path: "{{ download_location }}"
        state: directory
      loop: "{{ (
          bom_media | 
          flatten(levels=1) | 
          selectattr('override_target_location', 'defined') | 
          map(attribute='override_target_location') | 
          list
        ) + [default_location] | 
        unique }}"
      loop_control:
        loop_var: download_location
```